pkg/test/util/retry/retry.go

```diff
@@ -182,38 +182,59 @@ func UntilComplete(fn RetriableFunc, options ...Option) (any, error) {
 		default:
 		}
 
-		result, completed, err := fn()
-		attempts++
-		if completed {
-			if err == nil {
-				successes++
-			} else {
-				successes = 0
-			}
-			if successes >= cfg.converge {
-				return result, err
-			}
-
-			// Skip delay if we have a success
-			continue
-		}
-		successes = 0
-		if err != nil {
-			scope.Debugf("encountered an error on attempt %d: %v", attempts, err)
-			lasterr = err
-		}
-
+		done := make(chan struct {
+			result    any
+			completed bool
+			err       error
+		}, 1)
+
+		go func() {
+			result, completed, err := fn()
+			done <- struct {
+				result    any
+				completed bool
+				err       error
+			}{result, completed, err}
+		}()
```

Comment:
this will bring another issue, if the fn is blocked forever, we have no way to terminate it. And it is go routine leak